pkg/cmd/server/api/types.go

```diff
@@ -112,12 +112,20 @@ type NodeConfig struct {
 	// MasterClientConnectionOverrides provides overrides to the client connection used to connect to the master.
 	MasterClientConnectionOverrides *ClientConnectionOverrides
 
+	// DNSBindAddress is the ip:port to serve on. If this is not set, the DNS server will not be started.
+	DNSBindAddress string
```

Comment:
group DNSNameservers with this param and lay out the two setup scenarios really clearly